projects/term-finance/index.js

```diff
@@ -0,0 +1,66 @@
+const { request, gql } = require("graphql-request");
+const { getBlock } = require("../helper/http");
+
+const graphs = {
+  ethereum:
+    "https://api.studio.thegraph.com/query/33765/term-finance-mainnet/version/latest",
+};
+
+function addTokenBalance(balances, token, balance) {
+  if (!balances[token]) {
+    balances[token] = balance;
+  } else {
+    balances[token] = balances[token].add(balance);
+  }
+}
+
+function tvlPaged(chain) {
+  return async (_, _b, { [chain]: block }) => {
+    block = await getBlock(_, chain, { [chain]: block });
+    const balances = {};
+    const size = 1000;
+    const graphQueryPaged = gql`
+    query poolQuery($lastId: String, $block: Int) {
+      termRepoCollaterals(
+        block: { number: $block },
+        first: ${size},
+        where: {
+          id_gt: $lastId,
+          term__delisted: false
+        }
+      ) {
+        id
+        amountLocked
+        collateralToken
+      }
+    }`;
+
+    let lastId = "";
+    do {
+      const { termRepoCollaterals } = await request(
+        graphs[chain],
+        graphQueryPaged,
+        { lastId, block: block - 5000 }
+      );
+      for (const collateral of termRepoCollaterals) {
+        addTokenBalance(
+          balances,
+          collateral.collateralToken,
+          collateral.amountLocked
+        );
+      }
+      lastId = termRepoCollaterals[termRepoCollaterals.length - 1]?.id;
+    } while (termRepoCollaterals.length === size);
```

Comment:
don't understand this condition. Why does the length of collaterals have to equal 1000?

Reply:
@aazhou1 one is paging though list of items and page size is set as 1000, so when # items returned is less than 1000 it means, it is the last page